### PR TITLE
kv, stmt: add kv.Option, optimize table scan

### PR DESCRIPTION
**Pull request by disksing**
When doing table scan and failed get a key from cache, we launch a RangePrefetch to underlying storage instead of Get.

### CODE REVIEW

File: store/localstore/txn.go [hunk 1]
```diff
@@ -278,3 +279,18 @@ func (txn *dbTxn) LockKeys(keys ...kv.Key) error {
 	}
 	return nil
 }
+
+func (txn *dbTxn) EnableOption(opt kv.Option, val interface{}) {
```
Comment (ngaut) [inline store/localstore/txn.go]: s/EnableOption/SetOption

File: store/localstore/txn.go [hunk 2]
```diff
@@ -278,3 +279,18 @@ func (txn *dbTxn) LockKeys(keys ...kv.Key) error {
...
+	txn.opts[opt] = val
+}
+
+func (txn *dbTxn) DisableOption(opt kv.Option) {
```
Comment (ngaut) [inline store/localstore/txn.go]: s/DisableOption/DelOption

File: kv/kv.go [hunk 3]
```diff
@@ -96,6 +96,23 @@ type EncodeFn func(raw interface{}) (interface{}, error)
...
+	// Get gets an option value.
+	Get(opt Option) (v interface{}, ok bool)
+}
+
+const (
+	// RangePrefetchOnCacheMiss directives that when dealing with a Get operation and failed to read data from cache,
+	// it will launch a RangePrefetch to underlying storage instead of Get. The range starts from requested key and
+	// has a limit of the option value. The feature is disabled if option value <= 0 or value type is not int.
+	// This option is particularly useful when we have to do sequential Gets, e.g. table scans.
+	RangePrefetchOnCacheMiss Option = 1 + iota
```
Comment (ngaut) [inline kv/kv.go]: s/1 + iota/iota + 1

**Comment by disksing** [comment]
PTAL @ngaut @c4pt0r

**Comment by c4pt0r** [comment]
LGTM

**Comment by ngaut** [comment]
LGTM

**Pull request merged**